crates/bevy_ui/src/render/mod.rs

```diff
@@ -618,13 +619,14 @@ pub fn extract_text_uinodes(
     >,
 ) {
     // TODO: Support window-independent UI scale: https://github.com/bevyengine/bevy/issues/5621
-    let scale_factor = windows
+
+    let scale_factor = (windows
         .get_single()
-        .map(|window| window.resolution.scale_factor())
-        .unwrap_or(1.0)
-        * ui_scale.0;
+        .map(|window| window.scale_factor())
+        .unwrap_or(1.)
+        * ui_scale.0) as f32;
```

Comment:
This would have worked. tbh re-using the same bind name would also be fine.
```rust
let window = windows.get_single();
let window_scale = window.map_or(1., |w| w.resolution.scale_factor());
let scale_factor = (window_scale * ui_scale.0) as f32;
```